prov/gni/src/gnix_dom.c

```diff
@@ -49,13 +49,130 @@ uint32_t gnix_def_gni_rx_cq_size = 16384;
 /* TODO: should we use physical pages for gni cq rings? This is a question for Zach */
 gni_cq_mode_t gnix_def_gni_cq_modes = GNI_CQ_PHYS_PAGES;
 
+static int gnix_cm_nic_free(struct gnix_cm_nic *cm_nic)
+{
+	int ret = FI_SUCCESS;
+	gni_return_t status;
+
+	if (cm_nic == NULL)
+		return -FI_EINVAL;
+
+	atomic_dec(&cm_nic->ref_cnt);
```

Comment:
Potential double free situation.  Save return value from `atomic_dec` to make sure only one thread cleans up.